leap_year.rb

```diff
@@ -0,0 +1,18 @@
+def leap_year
+    leap_years = Array.new
+    print "Please enter a year to start. "
+    first_year = gets.chomp.to_i
+    print "Please enter a year to end. "
+    end_year = gets.chomp.to_i 
+  while first_year <= end_year
+    if first_year % 400 == 0
+      leap_years.push(first_year) 
+     elsif first_year % 100 == 0
+     elsif first_year % 4 == 0
+      leap_years.push(first_year) 
+      end
```

Comment:
I can understand why it's split up into multiple `elsif`s, but the following solution with the logical operators `&&` and `||` operators is also valid if you're interested:

```suggestion
    if first_year % 4 == 400 || (first_year % 100 != 0 && first_year % 4 == 0)
      leap_years.push(year)
    end
```

There is also the solution provided by John McGarvey, which is slightly more performant because of the order in which the checks are done:

```suggestion
    if (first_year % 4 == 0 && first_year % 100 != 0) || first_year % 400 == 0
      leap_years.push(year)
    end
```